src/crate/client/sqlalchemy/compiler.py

```diff
@@ -71,10 +73,27 @@ def rewrite_update(clauseelement, multiparams, params):
 
 
 @sa.event.listens_for(sa.engine.Engine, "before_execute", retval=True)
-def crate_before_execute(conn, clauseelement, multiparams, params):
+def crate_before_execute(conn, clauseelement, multiparams, params, *args, **kwargs):
     is_crate = type(conn.dialect).__name__ == 'CrateDialect'
     if is_crate and isinstance(clauseelement, sa.sql.expression.Update):
-        return rewrite_update(clauseelement, multiparams, params)
+        if SA_VERSION >= SA_1_4:
+            if params is None:
+                multiparams = ([],)
+            else:
+                multiparams = ([params],)
+            params = {}
+
+        clauseelement, multiparams, params = rewrite_update(clauseelement, multiparams, params)
+
+        if SA_VERSION >= SA_1_4:
+            if multiparams[0]:
+                params = multiparams[0][0]
+            else:
+                params = multiparams[0]
+            multiparams = []
+
+        return clauseelement, multiparams, params
```

Comment:
very minor: Is this return here needed? looks like if falling through it would return the same.

Reply:
Thanks for spotting it, that slipped through. I've removed the redundant `return` statement with the original commit which accidentally introduced it.